ecs-agent/wsclient/client.go

```diff
@@ -114,6 +118,9 @@ type RequestHandler interface{}
 type RequestResponder interface {
 	// Name returns the name of the responder. This is used mostly for logging.
 	Name() string
+	// RegisterResponder registers a function that can be invoked in response
+	// to receiving and processing a websocket request message.
+	RegisterResponder(RespondFunc)
```

Comment:
We removed this method after some discussion. Please see https://github.com/aws/amazon-ecs-agent/pull/3724#discussion_r1212297411.

Reply:
Oh, thanks Amogh, I took a look and it looks reasonable to me. I'll try to use what we have already.